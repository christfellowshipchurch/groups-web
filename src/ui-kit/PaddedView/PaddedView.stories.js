import React from 'react';
import { storiesOf } from '@storybook/react';

import PaddedView from '.';

storiesOf('ui-kit/PaddedView', module)
  .add('default', () => <PaddedView>This text is in a PaddedView</PaddedView>)
  .add('horizontal (false)', () => (
    <PaddedView horizontal={false}>
      This text is in a PaddedView without horizontal padding
    </PaddedView>
  ))
  .add('vertical (false)', () => (
    <PaddedView vertical={false}>
      This text is in a PaddedView without vertical padding
    </PaddedView>
  ));
