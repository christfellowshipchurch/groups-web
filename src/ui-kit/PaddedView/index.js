// import React from 'react';
import styled from 'styled-components';

import PropTypes from 'prop-types';

const PaddedView = styled.div.attrs({
  className: 'ui-kit.PaddedView',
})`
  padding-top: ${({ theme, vertical }) =>
    vertical ? theme.sizing.baseUnit(1) : 0};
  padding-right: ${({ theme, horizontal }) =>
    horizontal ? theme.sizing.baseUnit(1) : 0};
  padding-bottom: ${({ theme, vertical }) =>
    vertical ? theme.sizing.baseUnit(1) : 0};
  padding-left: ${({ theme, horizontal }) =>
    horizontal ? theme.sizing.baseUnit(1) : 0};
`;

PaddedView.propTypes = {
  horizontal: PropTypes.bool,
  vertical: PropTypes.bool,
};

PaddedView.defaultProps = {
  horizontal: true,
  vertical: true,
};

export default PaddedView;
