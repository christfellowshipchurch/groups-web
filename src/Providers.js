import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
