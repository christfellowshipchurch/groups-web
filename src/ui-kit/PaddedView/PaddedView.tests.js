import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../Providers';

import PaddedView from '.';

describe('the PaddedView component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <PaddedView />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render without horizontal padding', () => {
    const tree = renderer.create(
      <Providers>
        <PaddedView horizontal={false} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render without vertical padding', () => {
    const tree = renderer.create(
      <Providers>
        <PaddedView vertical={false} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
