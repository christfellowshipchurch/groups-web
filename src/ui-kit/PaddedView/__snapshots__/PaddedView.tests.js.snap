// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PaddedView component should render 1`] = `
.c0 {
  padding-top: 1rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
  padding-left: 1rem;
}

<div
  className="c0 ui-kit.PaddedView"
/>
`;

exports[`the PaddedView component should render without horizontal padding 1`] = `
.c0 {
  padding-top: 1rem;
  padding-right: 0;
  padding-bottom: 1rem;
  padding-left: 0;
}

<div
  className="c0 ui-kit.PaddedView"
/>
`;

exports[`the PaddedView component should render without vertical padding 1`] = `
.c0 {
  padding-top: 0;
  padding-right: 1rem;
  padding-bottom: 0;
  padding-left: 1rem;
}

<div
  className="c0 ui-kit.PaddedView"
/>
`;
